src/global_timestamp_reader.cpp

```diff
@@ -183,15 +213,15 @@ namespace librealsense
                 _min_command_delay = command_delay;
             }
             double system_time(system_time_finish - _min_command_delay);
-            if (sample_hw_time < _last_sample_hw_time)
+            std::lock_guard<std::recursive_mutex> lock(_read_mtx);
```

Comment:
I think that mutex shall be applied in all places the internal buffer is being evaluated. For review